airflow/decorators/base.py

```diff
@@ -550,11 +550,13 @@ def __attrs_post_init__(self):
         super(DecoratedMappedOperator, DecoratedMappedOperator).__attrs_post_init__(self)
         XComArg.apply_upstream_relationship(self, self.op_kwargs_expand_input.value)
 
-    def _expand_mapped_kwargs(self, context: Context, session: Session) -> tuple[Mapping[str, Any], set[int]]:
+    def _expand_mapped_kwargs(
+        self, context: Context, session: Session, *, include_xcom: bool = True
```

Comment:
Since this is internal code, I think we should make `include_xcom` a required argument so it’s always a conscious decision to include xcom or not. It is too easy to get this wrong otherwise (especially since the default—True—is insecure).

Reply:
Same applies to `resolve`.

Reply:
Makes sense. Just updated it